### PR TITLE
feat(DLI): add new resource: dli sql job

### LINKED ISSUE #1603: [DLI]Need a new resource to manage SQL jobs
For DLI service, we need a new resource to manage SQL jobs.

### PR DESCRIPTION
**Pull request by hmbig2**
**What this PR does / why we need it**:
feat(DLI): add new resource: dli sql job

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #1603 

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed
```
make testacc TEST='./huaweicloud/services/acceptance/dli' TESTARGS='-run=TestAccResourceDliSqlJob_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run=TestAccResourceDliSqlJob_basic -timeout 360m -parallel 4
=== RUN   TestAccResourceDliSqlJob_basic
=== PAUSE TestAccResourceDliSqlJob_basic
=== CONT  TestAccResourceDliSqlJob_basic
--- PASS: TestAccResourceDliSqlJob_basic (41.22s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       41.305s
```

```
make testacc TEST='./huaweicloud/services/acceptance/dli' TESTARGS='-run=TestAccResourceDliSqlJob_query'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/dli -v -run=TestAccResourceDliSqlJob_query -timeout 360m -parallel 4
=== RUN   TestAccResourceDliSqlJob_query
=== PAUSE TestAccResourceDliSqlJob_query
=== CONT  TestAccResourceDliSqlJob_query
--- PASS: TestAccResourceDliSqlJob_query (41.37s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/dli       41.493s
```

### CODE REVIEW

**Comment by hmbig2** [comment]
review after 

 #1621

File: docs/resources/dli_sqljob.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+  database_name = var.database_name
+  queue_name    = var.queue_name
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String, ForceNew) Specifies the region in which to create the dli table resource. If omitted,
```
Comment (Lance52259) [inline docs/resources/dli_sqljob.md]: for all service shorthands, we use uppercase strings to represent them.

File: docs/resources/dli_sqljob.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+ Changing this parameter will create a new resource.
+
+* `tags` - (Optional, Map, ForceNew) Specifies babel of a Job. Changing this parameter will create a new resource.
+
+* `conf` - (Optional, List, ForceNew) Specifies the configuration parameters for the SQL job. Changing this parameter
+ will create a new resource. Structure is documented below.
+
+ The `conf` block supports:
+
+   * `spark_sql_max_records_per_file` - (Optional, Int, ForceNew) Maximum number of records to be written
```
Comment (Lance52259) [inline docs/resources/dli_sqljob.md]: missing description of ForceNew.

File: docs/resources/dli_sqljob.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+* `sql` - (Required, String, ForceNew) Specifies SQL statement that you want to execute.
+  Changing this parameter will create a new resource.
+
+* `database_name` - (Required, String, ForceNew) Specifies the database where the SQL is executed. This argument does
+ not need to be configured during database creation. Changing this will create a new resource.
+
+* `queue_name` - (Optional, String, ForceNew) Specifies queue which this job to be submitted belongs.
+ Changing this parameter will create a new resource.
+
+* `tags` - (Optional, Map, ForceNew) Specifies babel of a Job. Changing this parameter will create a new resource.
```
Comment (Lance52259) [inline docs/resources/dli_sqljob.md]: `label` ?

File: docs/resources/dli_sqljob.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+
+* `conf` - (Optional, List, ForceNew) Specifies the configuration parameters for the SQL job. Changing this parameter
+ will create a new resource. Structure is documented below.
+
+ The `conf` block supports:
+
+   * `spark_sql_max_records_per_file` - (Optional, Int, ForceNew) Maximum number of records to be written
+    into a single file. If the value is zero or negative, there is no limit. Default value is `zero`.
```
Comment (Lance52259) [inline docs/resources/dli_sqljob.md]: ``Default value is `0`.`` or maybe ``Default value is  zero.`` ?

File: docs/resources/dli_sqljob.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+    Default value is `209715200`.
+
+   -> Currently, only the configuration unit metastore table that runs the ANALYZE TABLE COMPUTE statistics noscan
+    command and the file-based data source table that directly calculates statistics based on data files are supported.
+
+   * `spark_sql_shuffle_partitions` - (Optional, Int, ForceNew) Default number of partitions used to filter
+    data for join or aggregation. Default value is `4096`.
+
+   * `spark_sql_dynamic_partition_overwrite_enabled` - (Optional, Int, ForceNew) In dynamic mode, Spark does not delete
+    the previous partitions and only overwrites the partitions without data during execution.  Default value is `false`.
```
Comment (Lance52259) [inline docs/resources/dli_sqljob.md]: too many blank spaces between `execution.` and `Default`.

File: docs/resources/dli_sqljob.md [hunk 6]
```diff
@@ -0,0 +1,109 @@
...
+
+   * `spark_sql_files_max_partition_bytes` - (Optional, Int, ForceNew) Maximum number of bytes to be packed into a
+    single partition when a file is read. Default value is `134217728`.
+
+   * `spark_sql_bad_records_path` - (Optional, Int, ForceNew) Path of bad records.
+
+   * `dli_sql_sqlasync_enabled` - (Optional, Int, ForceNew) Specifies whether DDL and DCL statements are executed
+    asynchronously. The value true indicates that asynchronous execution is enabled.  Default value is `false`.
```
Comment (Lance52259) [inline docs/resources/dli_sqljob.md]: ditto.

File: docs/resources/dli_sqljob.md [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+
+## Timeouts
+
+This resource provides the following timeouts configuration options:
+
+* `Delete` - Default is 45 minute.
+
+## Import
+
+DLI SQL job can be imported by  `id`. For example,
```
Comment (Lance52259) [inline docs/resources/dli_sqljob.md]: too many blank spaces between `by` and `id`.

File: docs/resources/dli_sqljob.md [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+
+```
+terraform import huaweicloud_dli_sqljob.example  7f803d70-c533-469f-8431-e378f3e97123
```
Comment (Lance52259) [inline docs/resources/dli_sqljob.md]: ditto.

File: docs/resources/dli_sql_job.md [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+
+* `owner` - User who submits a job.
+
+* `job_type` - Type of a job, Includes **DDL**, **DCL**, **IMPORT**, **EXPORT**, **QUERY**, **INSERT**,
+ **DATA_MIGRATION**, **UPDATE**, **DELETE**, **RESTART_QUEUE** and **SCALE_QUEUE**.
+
+* `status` - Status of a job, including **RUNNING**, **SCALING**, **LAUNCHING**, **FINISHED**, **FAILED**,
+  and **CANCELLED.**
+
+* `start_time` - Time when a job is started. The timestamp is expressed in milliseconds.
```
Comment (ShiChangkuo) [inline docs/resources/dli_sql_job.md]: suggest to convert the value to a human readable format: 2021-10-29 17:00:00

File: huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_sql_job_test.go [hunk 10]
```diff
@@ -0,0 +1,184 @@
...
+}
+
+resource "huaweicloud_dli_table" "test" {
+  database_name      = huaweicloud_dli_database.test.name
+  name               = "acc_test_table"
+  data_location      = "DLI"
+  description        = "dli table test"
+
+  columns {
+    name = "name"
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_sql_job_test.go]: format error

File: huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_sql_job_test.go [hunk 11]
```diff
@@ -0,0 +1,184 @@
...
+resource "huaweicloud_dli_database" "test" {
+  name                  = "%s"
+  description           = "For terraform acc test"
+}
+
+resource "huaweicloud_dli_table" "test" {
+  database_name      = huaweicloud_dli_database.test.name
+  name               = "%s"
+  data_location      = "DLI"
+  description        = "dli table test"
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/dli/resource_huaweicloud_dli_sql_job_test.go]: too many spaces

File: huaweicloud/services/dli/huaweicloud_dli_sql_job.go [hunk 12]
```diff
@@ -0,0 +1,330 @@
...
+		return fmtp.DiagErrorf("Error creating DLI sql job: %s", rst.Message)
+	}
+
+	d.SetId(rst.JobId)
+	d.Set("schema", rst.Schema)
+	d.Set("rows", rst.Rows)
+	return resourceSqlJobRead(ctx, d, meta)
+}
+
+func resourceSqlJobRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```
Comment (ShiChangkuo) [inline huaweicloud/services/dli/huaweicloud_dli_sql_job.go]: suggest to use _ context.Context to indicate the param is not used

File: huaweicloud/services/dli/huaweicloud_dli_sql_job.go [hunk 13]
```diff
@@ -0,0 +1,330 @@
...
+	timeout time.Duration) error {
+	stateConf := &resource.StateChangeConf{
+		Pending: []string{"Pending"},
+		Target:  []string{"Done"},
+		Refresh: func() (interface{}, string, error) {
+			jobStatus, err := sqljob.Status(client, id)
+			if err == nil {
+				if jobStatus.Status == sqljob.JobStatusCancelled ||
+					jobStatus.Status == sqljob.JobStatusFinished ||
+					jobStatus.Status == sqljob.JobStatusFailed {
```
Comment (ShiChangkuo) [inline huaweicloud/services/dli/huaweicloud_dli_sql_job.go:309]: suggest to return an error when the Status is **JobStatusFailed**

File: huaweicloud/services/dli/huaweicloud_dli_sql_job.go [hunk 14]
```diff
@@ -0,0 +1,330 @@
...
+		cancelRst, cancelErr := sqljob.Cancel(client, jobId)
+		if cancelErr != nil {
+			return fmtp.DiagErrorf("cancel DLI sql job failed. %q:%s", jobId, dErr)
+		}
+		if cancelRst != nil && !cancelRst.IsSuccess {
+			return fmtp.DiagErrorf("cancel DLI sql job failed. %q:%s", jobId, dErr)
+		}
+
+	}
+	d.SetId("")
```
Comment (ShiChangkuo) [inline huaweicloud/services/dli/huaweicloud_dli_sql_job.go]: we should set the ID to nil at last.

File: huaweicloud/services/dli/huaweicloud_dli_sql_job.go (hunk 13 above)
Comment (hmbig2) [reply huaweicloud/services/dli/huaweicloud_dli_sql_job.go:309]: SQL job just a log. so `JobStatusFailed` is a finished status.

File: docs/resources/dli_sql_job.md (hunk 9 above)
Comment (hmbig2) [reply docs/resources/dli_sql_job.md]: OK，return with timezero in RFC-3339 format. e.g. `2019-10-12T07:20:50.52Z`

**Pull request merged**